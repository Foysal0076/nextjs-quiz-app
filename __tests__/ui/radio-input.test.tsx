import Radio from '@/components/ui/radio-input'
import { render, screen, fireEvent } from '@testing-library/react'

describe('Radio Component', () => {
  it('renders radio input correctly', () => {
    render(<Radio />)
    const radio = screen.getByRole('radio')
    expect(radio).toBeInTheDocument()
  })

  it('renders with label when provided', () => {
    render(<Radio label='Choose option' id='test-radio' />)
    const label = screen.getByText('Choose option')
    expect(label).toBeInTheDocument()
    expect(label).toHaveAttribute('for', 'test-radio')
  })

  it('can be checked and unchecked', () => {
    render(<Radio defaultChecked />)
    const radio = screen.getByRole('radio')

    expect(radio).toBeChecked()
  })

  it('can be checked when not checked', () => {
    render(<Radio />)
    const radio = screen.getByRole('radio')

    expect(radio).not.toBeChecked()
  })

  it('applies custom className correctly', () => {
    const customClass = 'custom-radio'
    render(<Radio className={customClass} />)
    const radio = screen.getByRole('radio')
    expect(radio).toHaveClass(customClass)
  })

  it('forwards ref correctly', () => {
    const ref = jest.fn()
    render(<Radio ref={ref} />)
    expect(ref).toHaveBeenCalled()
  })

  it('handles disabled state', () => {
    render(<Radio disabled />)
    const radio = screen.getByRole('radio')
    expect(radio).toBeDisabled()
  })

  it('handles name attribute', () => {
    render(<Radio name='test-group' />)
    const radio = screen.getByRole('radio')
    expect(radio).toHaveAttribute('name', 'test-group')
  })

  // Snapshot test
  it('matches snapshot', () => {
    const { container } = render(
      <Radio label='Test Option' id='test-radio' name='test-group' />
    )
    expect(container).toMatchSnapshot()
  })
})
