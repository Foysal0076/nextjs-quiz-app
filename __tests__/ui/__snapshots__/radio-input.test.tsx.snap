// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Component matches snapshot 1`] = `
<div>
  <div
    class="flex items-center space-x-2"
  >
    <input
      class="h-4 w-4 border-gray-300 bg-gray-100 text-primary-600 focus:ring-2 focus:ring-primary-500 dark:border-gray-600 dark:bg-gray-700 dark:checked:bg-primary-500 dark:focus:ring-primary-600"
      id="test-radio"
      name="test-group"
      type="radio"
    />
    <label
      class="cursor-pointer select-none text-sm text-neutral-700"
      for="test-radio"
    >
      Test Option
    </label>
  </div>
</div>
`;
